dao/generic.js

```diff
@@ -7,83 +7,184 @@ class GenericDAO {
+                stack: error.stack
+            });
+            throw new Error(`Failed to update ${this.model.name}`);
         }
-        // Accidental _id or id fields should be excluded.
-        const { _id, id: dataId, ...updateData } = data;
-        const res = await this.model.update({ where: { id }, data: updateData });
-        return { ...res, _id: res.id };
     }
 
     async updateMany(condition, data){
-        return await this.model.updateMany({ where: { ...condition }, data: { ...data }});
+        try {
+            return await this.model.updateMany({ where: { ...condition }, data: { ...data }});
+        } catch (error) {
+            console.error(`GenericDAO.updateMany failed for ${this.model.name}:`, {
+                error: error.message,
+                conditionKeys: condition && typeof condition === 'object' ? Object.keys(condition) : null,
+                dataKeys: data && typeof data === 'object' ? Object.keys(data) : null,
+                stack: error.stack
+            });
+            throw new Error(`Failed to update many ${this.model.name}`);
+        }
     }
 
     async deleteMany(where) {
-        return await this.model.deleteMany({ where});
+        try {
+            return await this.model.deleteMany({ where});
+        } catch (error) {
+            console.error(`GenericDAO.deleteMany failed for ${this.model.name}:`, {
+                error: error.message,
+                where: JSON.stringify(where),
+                stack: error.stack
+            });
+            throw new Error(`Failed to delete many ${this.model.name}`);
+        }
     }
 
     async delete(id) {
-        return await this.model.delete({ where: { id } });
+        try {
+            return await this.model.delete({ where: { id } });
+        } catch (error) {
+            console.error(`GenericDAO.delete failed for ${this.model.name}:`, {
+                error: error.message,
+                id,
+                stack: error.stack
+            });
+            throw new Error(`Failed to delete ${this.model.name}`);
+        }
     }
+
     /**
-     * Counts the number of documents in the collection based on the given filter and optional distinct fields.
+     * Counts the number of documents in the collection based on the given filter.
      *
      * @param {Object} where - The filter conditions to apply (e.g., { status: 'SUBMITTED' }).
-     * @param {string|string[]} distinct - A single field or an array of fields to count distinct values for.
-     * @returns {Promise<number>} - The count of matching documents (optionally distinct).
+     * @returns {Promise<number>} - The count of matching documents.
      */
-    async count(where, distinct) {
-        const arr = !Array.isArray(distinct) ? [distinct] : distinct;
-        const res = await this.model.findMany({
-            where,
-            distinct: arr
-        });
-        return res?.length || 0;
+    async count(where) {
```

Comment:
Could you roll back these changes? The async **countNodesBySubmissionID**(submissionID) function is using this code, so it needs to be either fixed or reverted.
**async count(where, distinct) {
        const arr = !Array.isArray(distinct) ? [distinct] : distinct;
        const res = await this.model.findMany({
            where,
            distinct: arr
        });
        return res?.length || 0**